packages/elements/src/dialog/index.ts

```diff
@@ -190,7 +195,7 @@ export class Dialog extends Overlay {
     const shouldUpdate = super.shouldUpdate(changedProperties);
 
     return shouldUpdate
-      || ((changedProperties.has('draggable') || changedProperties.has('header') || changedProperties.has(TranslatePropertyKey)) && this.opened);
+      || ((changedProperties.has('draggable') || changedProperties.has('header') || changedProperties.has('noInteractionLock') || changedProperties.has(TranslatePropertyKey)) && this.opened);
```

Comment:
Hi @goremikins, do you think `no-interaction-lock` should be able to toggle  ?

Reply:
I think it should not be here, noInteractionLock is handled in the overlay itself within shouldUpdate -> setRegisters. Overlays' `shouldUpdate` code is called from the dialog whenever it gets updated.

Reply:
@bualoy-napat can you take a look at this one

Reply:
@goremikins I have already logged the result from `shouldUpdate` from Overlay, it did not notice the change of `no-interaction-lock`.

Reply:
@goremikins Thank you for your suggestion. I have moved it to  `attributeChangedCallback`.